pkg/spiffe/spiffe.go

```diff
@@ -10,14 +10,19 @@ import (
 const (
 	Scheme = "spiffe"
 
+	URIPrefix = Scheme + "://"
+
 	// The default SPIFFE URL value for trust domain
 	defaultTrustDomain = "cluster.local"
 )
 
-var trustDomain = defaultTrustDomain
+var (
+	trustDomain = defaultTrustDomain
+)
 
 func SetTrustDomain(value string) {
-	trustDomain = value
+	// Replace special characters in spiffe
+	trustDomain = strings.Replace(value, "@", ".", -1)
```

Comment:
There doesn't seem to be a test for this?

Reply:
updated the test to account for this